core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt

```diff
@@ -151,3 +173,112 @@ class OnlineMultiplayer {
         return GameSaver.gameInfoPreviewFromString(Gzip.unzip(zippedGameInfo))
     }
 }
+
+/**
+ * Used to communicate data access between players
+ */
+class LockFile {
+    // The lockData is necessary to make every LockFile unique
+    // If Dropbox gets a file with the same content and overwrite set to false, it returns no
+    // error even though the file was not uploaded as the exact file is already existing
+    var lockData = UUID.randomUUID().toString()
+}
+
+/**
+ *	Wrapper around OnlineMultiplayer's synchronization facilities.
+ *
+ *	Based on the design of Mutex from kotlinx.coroutines.sync, except that when it blocks,
+ *	  it blocks the entire thread for an increasing period of time via Thread.sleep()
+ */
+class ServerMutex(val gameInfo: GameInfoPreview) {
+    private var locked = false
```

Comment:
Personally I found "`locked`" to be just a little confusing, as I expected it to be more global, I.E. "Is this lock currently held by *any* client?/Does this file exist on DropBox?", but it looks to be more like "Is this client currently holding the lock?"